src/modules/verifiable-credentials/verifiable-credentials.types.ts

```diff
@@ -0,0 +1,97 @@
+import { TypedDataField } from '@ethersproject/abstract-signer';
+import {
+  ICredentialSubject,
+  IJsonLdCredential,
+  IPresentation,
+  IPresentationDefinition,
+} from '@sphereon/pex';
+
+export interface IssuerFields {
+  key: string;
+  value: string | number;
+}
+export interface CreateCredentialParams {
+  subject: string;
+  namespace: string;
+  version: string;
+  issuerFields?: IssuerFields[];
+}
```

Comment:
I would move these types, as well as `EWFCredentialSubject` to a separate `role-credential.types.ts` file. This can keep `verifiable-credentials.types.ts` to be more generic and organized.

I also think that `CreateCredentialParams` should be `RoleCredentialSubjectParams` as these aren't params for a general credential.

I think we should also just consider using the `EWFCredentialSubject`/`RoleCredentialSubject` directly and removing the separate params interface as they have the same fields essentially. What do you think of this?

```suggestion
export interface IssuerFields {
  key: string;
  value: string | number;
}
/*
 * Parameters required to construct the subject for a role credential
 * https://www.w3.org/TR/vc-data-model/#credential-subject
 */
export interface RoleCredentialSubjectParams {
  /*
   * https://www.w3.org/TR/vc-data-model/#identifiers
   */
  id: string;
  namespace: string;
  version: string;
  issuerFields?: IssuerFields[];
}
```

Reply:
These types will be moved to `ew-did-registry/credential-interface`

Reply:
@Harasz I don't think we should move types related specifically to role credentials to `ew-did-registry/credential-interface`. I think `ew-did-registry` should remain generic "SSI", not having EW specific details. IMO, it would be better to have these types, as they relate to role credentials, somewhere in [ew-credentials](https://github.com/energywebfoundation/ew-credentials/tree/develop/packages)
Would you agree with this @JGiter @nichonien ?

Reply:
Yeah, sorry. I thought they're already in `@ew-did-registry/credential-interface`. I'll move it to separate file then

Reply:
As `IRole` is defined in `iam-client-lib` I think types related to `IRole` also should be defined in `iam-client-lib`